ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -255,6 +258,34 @@ func (r *RayJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&rayv1alpha1.RayJob{}).
 		Owns(&rayv1alpha1.RayCluster{}).
 		Owns(&corev1.Service{}).
+		WithEventFilter(predicate.Funcs{
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				r.Log.Info("event to delete", "event", e)
+
+				job, ok := e.Object.(*rayv1alpha1.RayJob)
+				if !ok {
+					r.Log.Info("failed to get job object from event", "event", e)
+					return false
+				}
+
+				if job.Status.JobStatus == rayv1alpha1.JobStatusRunning || job.Status.JobStatus == rayv1alpha1.JobStatusPending {
+					if job.Status.DashboardURL == "" || job.Status.JobId == "" {
+						r.Log.Info("dashboardURL or job_id is empty", "job", job)
```

Comment:
In this case, it means the job has not been submitted to the ray cluster. We don't need to do anything, right?

Reply:
yes, there would be some corner case.
if the dashboard url and the job id is not provided, we can not stop these job